sharding/test/trivial_integration_test.go

```diff
@@ -142,3 +143,38 @@ func TestErrorsIfShardingKeyChanged(t *testing.T) {
 	assert.NotNil(t, errorHandler.LastError)
 	assert.Equal(t, "sharding key changed from 2 to 1", errorHandler.LastError.Error())
 }
+
+type InsertDDLWriter struct {
+	db *sql.DB
+}
+
+func (this *InsertDDLWriter) Run() {
+	this.db.Exec("ALTER TABLE gftest.table1 ADD COLUMN (bar INT)")
+}
+
+func (this *InsertDDLWriter) Stop() {}
+
+func (this *InsertDDLWriter) Wait() {}
+
+func (this *InsertDDLWriter) SetDB(db *sql.DB) {
+	this.db = db
+}
+
+func TestErrorsIfDDLDetected(t *testing.T) {
```

Comment:
The Golang integration testing environment is theoretically "deprecated" and we're supposed to use the ruby one, even tho that one is also imperfect. Is there a reason to have the test here as opposed to be in `test/integration`? Does this need to be a sharding-specific test?

Reply:
I didn't realise the Golang env is deprecated, I can rework this to be a Ruby test. I actually pinged you for review by mistake as this is still a draft, but this is good feedback, thanks 🙏